pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableFactory.java

```diff
@@ -31,8 +35,10 @@ public static DataTable getDataTable(ByteBuffer byteBuffer)
       throws IOException {
     int version = byteBuffer.getInt();
     switch (version) {
-      case 2:
-        return new DataTableImplV2(byteBuffer);
+      case VERSION_2:
+        return convertDataTableImplV2ToV3(new DataTableImplV2(byteBuffer));
```

Comment:
Since the implementation between V2 and V3 is decoupled with common parts abstracted into DataTableUtils, why do we need a converter ?

Reply:
We should simply call `new DataTableImplV2(byteBuffer)` or `new DataTableImplV3(byteBuffer)` depending on the version